docs/design/design.md

```diff
@@ -0,0 +1,22 @@
+# Design Documentation
+
+This is an attempt to document the OpenShift Installer's existing design principles and also articulate design goals to guide with new feature development and in some cases inform bug fixes.
+
+## Design Goals
+
+### Install methods
+The OpenShift Installer currently supports 2 installation types, UPI (User Provisioned Infrastructure) and IPI (Installer Provisionined Infrastructure). IPI is a highly opinionated method of installing OpenShift. The IPI installation method tries to automate a large portion of the OpenShift Install with the least amount of intervention from the user. While it tries to take care of supporting the most common install scenarios, it cannot be expected to support every install variation that a user wants to use for their cloud.
+
+This is where the UPI offers the greater flexibility required for these highly specialized installs. In this install method, the user is expected to provide and manage several non-OpenShift components needed for the install on their own. For this reason, documentation surrounding the UPI flow are especially important.
+
+The Openshift Installer has the responsibility to validate configuration provided to it. This validation step is common to both the UPI and IPI install methods.
+
+### Cluster Configuration
+
+It is a design goal for the Installer to obtain the simplest and most minimal configuration required to fully create an OpenShift cluster on the platform and with the capabilities requested by the user. So, for that reason, any information that can be learned from the install environment should not be requested as an input configuration during the install.
+
+The installer gathers all user configuration and consolidates it the install-config.yaml file and uses this configuration to generated a slew of manifests. The Installer bootstraps the cluster by starting several OpenShift operators and providing configuration to them via these manifests. These operators in turn use these manifests to generate the resulting cluster. It then becomes the responsibilty of these Operators to manage this configuation data and the resulting changes within the cluster on day-2. Installer does not have a role to play in managing this configuration on day-2.
```

Comment:
> The Installer bootstraps the cluster by starting several OpenShift operators and providing configuration to them via these manifests.

The bootstrap process is already described in the design overview doc, and I think the same point, which is that the installer lays down configuration for cluster operators, without going into this detail, which could be a little confusing.

Reply:
Tried rewording this.